### PR TITLE
[FIX] portal: fix issues introduced by the portal redesign

### PR DESCRIPTION
**Pull request by chgo-odoo**
This PR has two main objectives : 

## Sidebar layout
During the portal redesign, we decided to make the sidebar width adapts to its content. While this approach helped us to avoid layout issues, it was sometimes too much content dependant, meaning that when more when two buttons were displayed side by side it would look okay, while having only a single button would place the sidebar at the extreme opposite side of the invoice, making it floating into the layout.

This PR reverts this approach by applying a fixed width to the sidebar, and uses some padding utilities to prevent cluttered layouts.

**Note:** This PR only includes important fixes that either:

(1) make the user experience not as good as it should be ;
(2) are needed to ensure that new features/changes that are being made to the Customer Portal this year provide a good visual result.

Other PR's will follow to refine the design of the customer portal introduced in Odoo 17.

## Bugs, inconsistencies or forgotten views of the customer portal UI

When the portal redesign happened, an overlay of design was applied to the templates, but the templates themselves were not really revamped and/or cleaned. This created a lot of unused utility classes, useless HTML tags, etc.

<details><summary> List of all the changes</summary>
<p>

- #### portal, portal_*
  - Reinforce the proximity between the sidebar and the main content in order to make the actions easily reachable ;
  - Fix the page layout by adjusting the columns used by both elements ;
  - First clean of the sidebar buttons layout by cleaning useless elements in the XML ;
  - Fix design inconsistencies (avatar roundness, wrong button classes, unconsistent badges, weird content organization, ...) ; 
  - Reintroduce the bottom padding to the main container in order to avoid that the footer sticks to the content ;
  - Convert secondary buttons to light ;
  - Ensure that the design of elements can handle "dark mode" ;

- #### website_crm_partner_assign
  - Fix an issue about the view still using card classes ;
  - Fix an issue about some unnecessary HTML tags in the layout ;

- #### mrp_subcontracting
  - Fix an issue about the breadcrumbs not working correctly ; 
  - Fix an issue about the `<iframe>` having a weird design.

</p>
</details> 

## + Add the possibility for users to hide pictograms to align to their needs
While doing fixes to the Customer Portal, we take that opportunity to add an option within the editor so that users can choose whether they want to display these pictograms or not. This will eventually allows users to avoid using assets that are not aligned with their needs.

- task-3703251
- follow up of task-3046004

### CODE REVIEW

File: addons/portal/views/portal_templates.xml [hunk 1]
```diff
@@ -260,7 +260,7 @@
                         <span t-if="placeholder_count" t-att-class="'' if show_count else 'd-none'" t-att-data-placeholder_count="placeholder_count"/>
                         <span t-out="title"/>
                     </h5>
-                    <p class="m-0 text-600">
+                    <p class="m-0 opacity-50">
```
Comment (chgo-odoo) [inline addons/portal/views/portal_templates.xml:263]: Note: I'm not sure this is the right approach, it seems weird to have to apply an opacity on a text to ensure it is readable. Perhaps something to handle with  --o-cc variables (?)

File: addons/sale/views/sale_portal_templates.xml [hunk 2]
```diff
@@ -136,10 +136,10 @@
             <div class="row o_portal_sale_sidebar">
                 <!-- Sidebar -->
                 <t t-call="portal.portal_record_sidebar" id="sale_order_portal_sidebar">
-                    <t t-set="classes" t-value="'d-print-none col-lg-3 col-xl-4'"/>
+                    <t t-set="classes" t-value="'d-print-none col-lg-4 col-xl-3'"/>
 
                     <t t-set="title">
-                        <h2 t-field="sale_order.amount_total" data-id="total_amount"/>
+                        <h2 t-field="sale_order.amount_total" data-id="total_amount" class="text-break"/>
```
Comment (chgo-odoo) [inline addons/sale/views/sale_portal_templates.xml:142]: This is actually not recommended but as we fix the width of our columns, the text would either overflow or wrap. Not sure if another approach can be applied there

File: addons/account/views/account_portal_templates.xml [hunk 3]
```diff
@@ -122,14 +122,10 @@
...
                         <div class="d-flex flex-column gap-4">
-                            <div class="flex-basis-100 flex-basis-sm-50 flex-basis-lg-100 order-1 order-lg-0">
-                                <div class="o_download_pdf d-flex gap-1 flex-lg-column flex-xl-row">
-                                    <div class="flex-grow-1">
-                                        <a class="btn btn-secondary d-block o_download_btn" t-att-href="invoice.get_portal_url(report_type='pdf', download=True)" title="Download"><i class="fa fa-download"/> Download</a>
-                                    </div>
-                                    <div class="flex-grow-1">
-                                        <a class="btn btn-secondary d-block o_print_btn o_portal_invoice_print" t-att-href="invoice.get_portal_url(report_type='pdf')" id="print_invoice_report" title="Print" target="_blank"><i class="fa fa-print"/> Print</a>
-                                    </div>
+                            <div>
```
Comment (stefanorigano) [inline addons/account/views/account_portal_templates.xml:125]: Question: is this `<div>` necessary?

**Review by stefanorigano — CHANGES_REQUESTED** [review]
Hello @chgo-odoo , very good, thanks so much for your work 🙌
Some minor remarks for you, thanks. 


Anso, not related to this PR, but the pager navigation is broken for signature (click → error 500).
image
As said is not related to your changes and it's reproducible in master. If you can't solve it could you report it to the dev team? Thanks!

File: addons/account/views/account_portal_templates.xml (hunk 3 above)
Comment (chgo-odoo) [reply addons/account/views/account_portal_templates.xml:125]: There might be a better way ^^

As the parent has a gap-4, I use this div to be able to xpath INSIDE of it to avoid having that gap-4 when xpathing extra buttons, will look for something else !

File: addons/project/views/project_portal_project_task_templates.xml [hunk 4]
```diff
@@ -218,7 +218,7 @@
                         </div>
                     </t>
                 </t>
-                <div id="task_content" class="o_portal_content col-12 col-lg-9 col-xl-8">
+                <div id="task_content" class="o_portal_content col-12 col-lg-8 col-xl-9 ps-xl-4">
                     <div id="card">
                         <div id="card_header" data-anchor="true">
```
[the remark this answers is not in the document]
Comment (chgo-odoo) [reply addons/project/views/project_portal_project_task_templates.xml:223]: Yeah also the text was truncated, which looks weird 🤔

File: addons/website_crm_partner_assign/views/website_crm_partner_assign_templates.xml [hunk 5]
```diff
@@ -435,92 +435,83 @@
...
-                            <div class="row" t-if="lead.date_deadline">
-                                <label class="col-4 fw-bold">Expected Closing</label>
-                                <span class="col-8" t-field="lead.date_deadline" />
+            <div class="d-flex justify-content-between align-items-center mb-4">
+                <h4>
+                    Lead - <span t-field="lead.name"/>
+                </h4>
+                <span title="Rating" role="img" t-attf-aria-label="Rating: #{lead.priority} on 3">
+                    <t t-foreach="range(1, 4)" t-as="i">
+                        <span t-attf-class="fa fa-lg fa-star#{'' if i &lt;= int(lead.priority) else '-o'}"/>
```
[the remark this answers is not in the document]
Comment (chgo-odoo) [reply addons/website_crm_partner_assign/views/website_crm_partner_assign_templates.xml:444]: Looks okay to me, I will use this approach for other rating occurences

**Review by stefanorigano — CHANGES_REQUESTED** [review]
Hello @chgo-odoo . Thanks, good work here 👍 
Some remarks for you

File: addons/hr_timesheet/views/project_task_portal_templates.xml [hunk 6]
```diff
@@ -4,9 +4,7 @@
     <template id="portal_my_task" inherit_id="project.portal_my_task" name="Portal: My Task with Timesheets">
         <xpath expr="//div[@id='task-nav']" position="before">
             <div t-if="timesheets and allow_timesheets" class="d-grid flex-grow-1" id='nav-report'>
-                <div class="o_download_pdf">
```
[the remark this answers is not in the document]
Comment (chgo-odoo) [reply addons/hr_timesheet/views/project_task_portal_templates.xml:7]: It is only used within XML files, yes.

File: addons/account/views/account_portal_templates.xml [hunk 7]
```diff
@@ -144,7 +140,7 @@
                                     </small>
                                 </h6>
                                 <div class="o_portal_contact_details d-flex flex-column gap-2">
-                                    <div class="d-flex justify-content-start align-items-center gap-2">
+                                    <div class="d-flex align-items-center gap-2">
```
[the remark this answers is not in the document]
Comment (chgo-odoo) [reply addons/account/views/account_portal_templates.xml:143]: Thanks, I wasn't indeed taking into account the potential modifications that would be applied to the layout.

**Comment by stefanorigano** [comment]
Hello @chgo-odoo 
> Should we shorten the editor labels even more?

Nope, looks ok to me.  I'd suggest to send a msg in the POs channel to ensure that everyone is ok with the change 👍 

> About the alert-link class, I didn't apply it the class because that would mean applying a font-weight: 600 to the inner content,

I still prefer the alert-link "color" (again, for readability).
We can add an `fw-normal` to address the font-weight... actually, it would be good  to override user's headings customizations as well. I'd go further and remove the `<h5>` tag and using a fs-5 class 👍 

We can check the details tomorrow if needed 😉

**Comment by chgo-odoo** [comment]
Hello everyone, I've got a bit of a favor to ask here 😅 I've got this pull request that's a target for this freeze and I'd like to get your input on it.

I know things are crazy busy right now with everyone pushing to hit their targets and reviewing other PR's to help them reach their targets, but your knowledge would be super valuable here.

The PR is +/- 600 lines long, so it's a bit of a chunk I admit 😅 (the enterprise counterpart is way smaller) but most of the changes should be visual here (moving stuff around, improving some XML structures, changing some styling) so there shouldn't be anything functional changes or blocking point.

**No pressure at all, and I totally understand if you're unable to find the time**. Just thought I'd reach out and see if it might be possible to merge this.

Thanks !

File: addons/account_payment/views/account_portal_templates.xml [hunk 8]
```diff
@@ -55,19 +55,17 @@
...
         <xpath expr="//t[@t-call='portal.portal_record_sidebar']//h2[1]" position="after">
             <t t-set="pending_txs" t-value="invoice.transaction_ids.filtered(lambda tx: tx.state in ('pending', 'authorized') and tx.provider_code not in ('none', 'custom'))"/>
-            <div class="d-grid">
-                <div t-if="invoice.payment_state in ('paid', 'in_payment')" class="position-sm-absolute position-lg-relative me-auto me-lg-0 top-0 end-0 rounded text-bg-success fs-6 fw-normal badge" >
-                    <i class="fa fa-fw fa-check-circle"/> Paid
-                </div>
-                <div t-elif="pending_txs" class="position-sm-absolute position-lg-relative me-auto me-lg-0 top-0 end-0 rounded text-bg-info fw-normal fs-6 badge" >
-                    <i class="fa fa-fw fa-check-circle"/> Pending
-                </div>
+            <div t-if="invoice.payment_state in ('paid', 'in_payment')" class="rounded text-bg-success fw-normal badge" >
```
Comment (rugo-odoo) [inline addons/account_payment/views/account_portal_templates.xml:58]:
```suggestion
            <div t-if="invoice.payment_state in ('paid', 'in_payment')" class="rounded text-bg-success fw-normal badge">
```

File: addons/account_payment/views/account_portal_templates.xml [hunk 9]
```diff
@@ -55,19 +55,17 @@
...
+                <i class="fa fa-fw fa-check-circle"/> Paid
+            </div>
+            <div t-elif="pending_txs" class="position-sm-absolute position-lg-relative me-auto me-lg-0 top-0 end-0 rounded text-bg-info fw-normal fs-6 badge" >
```
Comment (rugo-odoo) [inline addons/account_payment/views/account_portal_templates.xml:61]:
```suggestion
            <div t-elif="pending_txs" class="position-sm-absolute position-lg-relative me-auto me-lg-0 top-0 end-0 rounded text-bg-info fw-normal fs-6 badge">
```

File: addons/account/views/account_portal_templates.xml [hunk 10]
```diff
@@ -111,22 +111,20 @@
...
                             <span t-else="1" t-field="invoice.amount_total"/>
                         </h2>
                         <div class="small" t-if="invoice.payment_state not in ('paid', 'in_payment') and invoice.move_type == 'out_invoice'"><i class="fa fa-clock-o"/><span class="o_portal_sidebar_timeago ml4" t-att-datetime="invoice.invoice_date_due"/></div>
                     </t>
 
                     <t t-set="entries">
-                        <div class="d-flex flex-column gap-4">
-                            <div class="flex-basis-100 flex-basis-sm-50 flex-basis-lg-100 order-1 order-lg-0">
-                                <div class="o_download_pdf d-flex gap-1 flex-lg-column flex-xl-row">
-                                    <div class="flex-grow-1">
```
Comment (rugo-odoo) [inline addons/account/views/account_portal_templates.xml:127]: This seems to be missing.

File: addons/account_payment/views/account_portal_templates.xml (hunk 9 above)
Comment (rugo-odoo) [inline addons/account_payment/views/account_portal_templates.xml:61]: We remove a lot of classes from the `Paid` badge but we keep them all here. Is it normal ? They should be the same right ? I would expect this one to have (almost) the exact same classes.

**Review by rugo-odoo — COMMENTED** [review]
A few things, I spotted in the accounting part.

File: addons/account_payment/views/account_portal_templates.xml (hunk 9 above)
Comment (chgo-odoo) [reply addons/account_payment/views/account_portal_templates.xml:61]: Indeed, this one should align with others and use only `rounded text-bg-info fw-normal badge`

**Comment by ticodoo** [comment]
Seems fine to me for the logistics (purchase/mrp) parts. While testing I did notice a small UI bug that seems to also be from the 17.0 redesign for `mrp_subcontracting`

saas-16.4:
image

17.0+
image

The buttons seem to have lost their styling in this case. To see it you have to:

Open "Manufacturing Orders" > Open the order (WH/IN/0007) > click on the burger button (see img)
image

and then set a Quantity (see img) + click somewhere outside the field so that the value saves and the buttons show up

image

It's not a big deal to me so do with it as you want :shrug:

**Comment by chgo-odoo** [comment]
Hello @ticodoo, thanks for your review on this

I noticed that we also miss some icons (the navigation one). I'd prefer to address these issues later as they are not really blocking and they might need some investigation.

At first sight my guess would be that there is an issue with the manifest but to be investigated

**Review by Odoonan — APPROVED** [review]
Reviewing for SM, LGTM, did not find any issue while testing either, good job :)

**Comment by chgo-odoo** [comment]
Hello everyone 👋

Thanks for your time and review on the PR, really appreciate !

Since apparently the are no other remarks, could we r+ this?

Thanks again!

File: addons/mrp_subcontracting/views/subcontracting_portal_templates.xml [hunk 11]
```diff
@@ -19,10 +19,13 @@
     <template id="portal_my_home_menu_production" name="Portal layout : production menu entries" inherit_id="portal.portal_breadcrumbs" priority="25">
         <xpath expr="//ol[hasclass('o_portal_submenu')]" position="inside">
             <t t-if="title">
-                <li t-if="page_name == 'production' or production" t-attf-class="breadcrumb-item #{'active' if breadcrumbs_url else ''}">
+                <li t-if="page_name == 'production' or picking" t-attf-class="breadcrumb-item #{'active' if breadcrumbs_url else ''}">
```
Comment (Whenrow) [inline addons/mrp_subcontracting/views/subcontracting_portal_templates.xml:22]: why did you change this ?

Comment (chgo-odoo) [reply addons/mrp_subcontracting/views/subcontracting_portal_templates.xml:22]: The breadcrumb is currently broken in Master, doing that change seems to fix its behavior. Is there something I'm not aware of ?

**Review by Whenrow — APPROVED** [review]

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge
